src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1158,11 +1156,7 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
 
                         if (cancellationToken.CanBeCanceled)
                         {
-                            if (connector.CommandCts.IsCancellationRequested)
-                            {
-                                connector.CommandCts.Dispose();
-                                connector.CommandCts = new TimeoutCancellationTokenSourceWrapper(TimeSpan.FromSeconds(connector.Settings.CancellationTimeout));
-                            }
+                            connector.CommandCts.Reset(CancellationToken.None);
```

Comment:
nit: explicit CancellationToken.None not needed

Reply:
Yeah, no.
I opted for that because it's less visually impairing than the following:
```csharp
#pragma warning disable CA2016
                            connector.CommandCts.Reset();
#pragma warning restore CA2016
```

But you're right there's no need do always forward the ambient token just because a method you're calling happens to take a cancellation token with a default value.

Reply:
Ah, I see - I missed the warning part. If it's between the explicit CancellationToken.None and the warning, I'll go with the former for sure :rofl: